CMSgov/mint-app
# Easi 2034/beneficiaries

# EASI-2034

## Changes and Description

- Add Plan Beneficiaries Task List Backend



## How to test this change
1. Compare to Figma to make sure all fields are accounted for.
2.  Create a model plan and make sure that a Beneficiary gets created as expected
3. Update Plan Beneficiary
4. Confirm success when running the test suite.


To easily update plan beneficiary, you can run scripts/dev db:seed and update a seeded beneficiary as below

```gql

mutation updatePlanBeneficiaries #($input: PlanMilestonesInput!) 
{
    updatePlanBeneficiares (id: "b7aa2c9f-9e38-4e95-92d9-60bd791e8c59",changes: 
        {
     #Page 1
  beneficiaries: null
#   beneficiariesOther:
#   beneficiariesNote:
#   treatDualElligibleDifferent:
#   treatDualElligibleDifferentHow:
#   treatDualElligibleDifferentNote:
#   excludeCertainCharacteristics:
#   excludeCertainCharacteristicsCriteria:
#   excludeCertainCharacteristicsNote:
  #Page 2
  numberPeopleImpacted: 22
#   estimateConfidence:
  confidenceNote: "Confidence hooray!!!!"
#   beneficiarySelectionMethod:
#   beneficiarySelectionOther:

#   beneficiarySelectionNote:
#   #Page 3
#   beneficiarySelectionFrequency:
#   beneficiarySelectionFrequencyOther:
#   beneficiarySelectionFrequencyNote:
#   beneficiaryOverlap:
#   beneficiaryOverlapNote:
#   precedenceRules:


    }) {
        id
        modelPlanID

        beneficiaries
        beneficiariesOther
        beneficiariesNote
        treatDualElligibleDifferent
        treatDualElligibleDifferentHow
        treatDualElligibleDifferentNote
        excludeCertainCharacteristics
        excludeCertainCharacteristicsCriteria
        excludeCertainCharacteristicsNote
        numberPeopleImpacted
        estimateConfidence
        confidenceNote
        beneficiarySelectionMethod
        beneficiarySelectionOther
        beneficiarySelectionNote
        beneficiarySelectionFrequency
        beneficiarySelectionFrequencyOther
        beneficiarySelectionFrequencyNote
        beneficiaryOverlap
        beneficiaryOverlapNote
        precedenceRules        

        createdBy
        createdDts
        modifiedBy
        modifiedDts
        status
    }
}

```



## PR Author Review Checklist



## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.
